encoding.bs

```diff
@@ -1333,6 +1333,10 @@ dictionary TextDecoderOptions {
   boolean ignoreBOM = false;
 };
 
+dictionary TextDecoderOptionsWithLabel : TextDecoderOptions {
+  DOMString label = "utf-8";
```

Comment:
I wonder if we can get away with just supporting UTF-8 here. If you need more, you get the streaming API. Perhaps not even support `fatal` or `ignoreBOM`. Also makes it easier to implement without a lot of IDL bridging.